apps/req.c

```diff
@@ -1134,24 +1134,28 @@ static int auto_info(X509_REQ *req, STACK_OF(CONF_VALUE) *dn_sk,
         /*
          * Skip past any leading X. X: X, etc to allow for multiple instances
          */
-        for (p = v->name; *p; p++)
+        for (p = v->name; *p; p++){
+            int spec_char;
 #ifndef CHARSET_EBCDIC
-            if ((*p == ':') || (*p == ',') || (*p == '.')) {
+            spec_char = ((*p == ':') || (*p == ',') || (*p == '.'));
 #else
-            if ((*p == os_toascii[':']) || (*p == os_toascii[','])
-                || (*p == os_toascii['.'])) {
+            spec_char = ((*p == os_toascii[':']) || (*p == os_toascii[','])
+                    || (*p == os_toascii['.']));
 #endif
+            if (spec_char){
                 p++;
                 if (*p)
                     type = p;
                 break;
             }
+        }
+        int plus_char;
```

Comment:
still not a C feature.  Needs to be inside a block.  Suggest moving all variables to the start of the function just "to be safe" and avoid these problems.